packages/components/src/components/post-rating/post-rating.tsx

```diff
@@ -0,0 +1,179 @@
+import { Component, h, Event, EventEmitter, Prop, Host, State, Element } from '@stencil/core';
+import { version } from '../../../package.json';
+
+@Component({
+  tag: 'post-rating',
+  styleUrl: 'post-rating.scss',
+  shadow: true,
+})
+export class PostRating {
+  @Element() host: HTMLPostRatingElement;
+  /**
+   * The current rating value
+   */
+  @Prop({ mutable: true }) currentRating = 0;
+
+  /**
+   * The index of the currently hovered star
+   */
+  @State() hoveredIndex: number;
+
+  /**
+   * The number of stars in the rating
+   */
+  @Prop() readonly max?: number = 5;
+
+  /**
+   * Boolean for the disabled state of the component
+   */
+  @Prop() readonly disabled?: boolean = false;
+
+  /**
+   * If readonly is true, the component only displays a rating and is not interactive.
+   */
+  @Prop() readonly readonly?: boolean = false;
+
+  // prettier-ignore
+  /**
+   * Event emitted when the rating gets commited
+   */
+  @Event({
+    eventName: 'ratingChange',
```

Comment:
```suggestion
    eventName: 'change',
```

Yes it may show a warning in the terminal, but still works and is much closer to the standard.
Check if everything works as expeted, after changing this!